metrics/resource_view_test.go

```diff
@@ -0,0 +1,639 @@
+			// [new duration] in the future.
+			view.Unregister(globalCounter)
+			UnregisterResourceView(gaugeView, resourceCounter)
+			FlushExporter()
+
+			ocFake.srv.Stop() // Force close connections
+			ocFake.srv.GracefulStop()
+			records := []metricExtract{}
+			for record := range ocFake.published {
+				for _, m := range record.Metrics {
+					if len(m.Timeseries) > 0 {
+						labels := map[string]string{}
+						if record.Resource != nil {
+							labels = record.Resource.Labels
+						}
+						records = append(records, metricExtract{
+							Name:   m.MetricDescriptor.Name,
+							Labels: labels,
+							Value:  m.Timeseries[0].Points[0].GetInt64Value(),
+						})
+					}
+				}
+			}
+
+			if diff := cmp.Diff(expected, records, sortMetrics()); diff != "" {
+				t.Errorf("Unexpected OpenCensus exports (-want +got):\n%s", diff)
+			}
+		},
+	}, {
+		name: "Stackdriver",
+		init: func() error {
+			if err := initSdFake(&sdFake); err != nil {
+				return err
+			}
+			return UpdateExporter(configForBackend(stackdriver), logtesting.TestLogger(t))
+		},
+		validate: func(t *testing.T) {
+			records := []metricExtract{}
+			for record := range sdFake.published {
+				for _, ts := range record.TimeSeries {
+					name := ts.Metric.Type[len("custom.googleapis.com/knative.dev/testComponent/"):]
+					records = append(records, metricExtract{
+						Name:   name,
+						Labels: ts.Resource.Labels,
+						Value:  ts.Points[0].Value.GetInt64Value(),
+					})
+				}
+				if len(records) >= 4 {
+					// There's no way to synchronize on the internal timer used
+					// by metricsexport.IntervalReader, so shut down the
+					// exporter after the first report cycle.
+					FlushExporter()
+					sdFake.srv.GracefulStop()
+				}
+			}
+			if diff := cmp.Diff(expected, records, sortMetrics()); diff != "" {
+				t.Errorf("Unexpected Stackdriver exports (-want +got):\n%s", diff)
+			}
+		},
+	}}
+
+	for _, c := range harnesses {
+		t.Run(c.name, func(t *testing.T) {
+			sdFake.t = t
+			if err := c.init(); err != nil {
+				t.Fatalf("unable to init: %+v", err)
+			}
+
+			view.Register(globalCounter)
+			if err := RegisterResourceView(gaugeView, resourceCounter); err != nil {
+				t.Fatalf("Unable to register views: %v", err)
+			}
+			defer func() {
+				view.Unregister(globalCounter)
+				UnregisterResourceView(gaugeView, resourceCounter)
+			}()
+
+			for i, r := range resources {
+				ctx := context.Background()
+				Record(ctx, counter.M(int64(1)))
+				if r != nil {
+					ctx = metricskey.WithResource(ctx, *r)
+				}
+				Record(ctx, gauge.M(int64(i)))
+			}
+			c.validate(t)
+		})
+	}
+}
+
+func TestStackDriverExports(t *testing.T) {
+	sdFake := stackDriverFake{address: "localhost:12346", t: t}
+	eo := ExporterOptions{
+		Domain:    servingDomain,
+		Component: "autoscaler",
+		ConfigMap: map[string]string{
+			BackendDestinationKey:            string(stackdriver),
+			allowStackdriverCustomMetricsKey: "false",
```

Comment:
Could you add a test case that when `allowStackdriverCustomMetricsKey` is true, the built-in metrics are also supported? (The following assertions should still pass when flip `allowStackdriverCustomMetricsKey` from false to true)

Reply:
I just made a PR to evan's branch to cover this scenario.  https://github.com/evankanderson/knative-pkg/pull/8